server/player/player.go

```diff
@@ -2050,6 +2057,127 @@ func (p *Player) PunchAir() {
 	})
 }
 
+// MountEntity mounts the player to an entity if the entity is rideable and if there is a seat available.
+func (p *Player) MountEntity(r entity.Rideable) {
+	ctx := event.C()
+	p.handler().HandleMount(ctx, r)
+	ctx.Continue(func() {
+		if e, ok := r.(world.Entity); ok {
```

Comment:
Would check for this at the start of this method. Also maybe make `entity.Rideable` implement `world.Entity` to allow omission of this check entirely?